packages/components/src/tooltip/index.tsx

```diff
@@ -0,0 +1,106 @@
+/**
+ * External dependencies
+ */
+import * as Ariakit from '@ariakit/react/tooltip';
+
+/**
+ * WordPress dependencies
+ */
+import { useInstanceId } from '@wordpress/compose';
+import { Children } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import type { TooltipProps } from './types';
+import Shortcut from '../shortcut';
+import { positionToPlacement } from '../popover/utils';
+
+/**
+ * Time over anchor to wait before showing tooltip
+ *
+ * @type {number}
+ */
+export const TOOLTIP_DELAY = 700;
+
+function Tooltip( props: TooltipProps ) {
+	const { children, delay = TOOLTIP_DELAY, position, shortcut, text } = props;
+
+	const baseId = useInstanceId( Tooltip, 'tooltip' );
+	const describedById = text || shortcut ? baseId : undefined;
+
+	const isOnlyChild = Children.count( children ) === 1;
+	// console error if more than one child element is added
+	if ( ! isOnlyChild ) {
+		if ( 'development' === process.env.NODE_ENV ) {
+			// eslint-disable-next-line no-console
+			console.error(
+				'Tooltip should be called with only a single child element.'
+			);
+		}
+	}
+
+	const DEFAULT_PLACEMENT = 'bottom';
+
+	// Compute tooltip's placement:
+	// - compute it from the legacy `position` prop (if defined)
+	// - otherwise, fallback to the DEFAULT_PLACEMENT.
+	let computedPlacement;
+
+	if ( position !== undefined ) {
+		computedPlacement = positionToPlacement( position );
+	} else {
+		computedPlacement = DEFAULT_PLACEMENT;
+	}
+
+	const tooltipStore = Ariakit.useTooltipStore( {
+		// TODO: can remove the ignore once position has been fully deprecated
+		// Type error due to 'overlay' type from positionToPlacement
+		// @ts-ignore
+		placement: computedPlacement,
+		timeout: delay,
+	} );
+
+	return (
+		<>
+			<Ariakit.TooltipAnchor
+				onBlur={ () => tooltipStore.hide() }
+				store={ tooltipStore }
+				render={ isOnlyChild ? children : undefined }
+			>
+				{ ! isOnlyChild ? children : null }
+			</Ariakit.TooltipAnchor>
+			{ isOnlyChild && ( text || shortcut ) && (
+				<Ariakit.Tooltip
+					className="components-tooltip"
+					gutter={ 4 }
+					// 	hide tooltip when interacting with its anchor to match legacy tooltip
+					hideOnInteractOutside={ () => {
```

Comment:
This allows us match [the behaviour of our legacy tooltip](https://user-images.githubusercontent.com/35543432/260192090-e0840e85-9335-44cd-a2ff-1c867497c427.gif) so that the tooltip is hidden when the anchor is clicked. I don't believe this is the correct usage of [`hideOnInteractOutside`](https://ariakit.org/reference/tooltip#hideoninteractoutside), so if there is another way, that would be preferable. Perhaps @diegohaz has a suggestion for this? 

We did try adding this behavior to a `div` wrapper but that caused styles to break in existing use cases of our legacy tooltip.

Reply:
Yeah, I don't think `hideOnInteractOutside` is the right place to put that logic. I'd put it in the `onClick` handler on the tooltip anchor:

```jsx
<Ariakit.TooltipAnchor onClick={tooltipStore.hide} />
```

Reply:
Thank you!